settings.php

```diff
@@ -90,6 +90,13 @@
         get_string('uniquekeynewidtomaintain_desc', 'tool_mergeusers'),
         1));
 
-    // Add settings
+    // Searh by Profile fields.
+    $profilefieldsoptions = tool_mergeusers_build_profilefields_options();
```

Comment:
This setting calculates all options always. Settings like this should be made lazy: I mean, load options only when this setting is really shown by a user.

One way is this: https://github.com/jpahullo/moodle/commit/9c4510a358cdcd192085d8d96c99acc571292dac. But look at some other admin setting that loads options only whenever used, with the same idea: is there no option already? calculate and return. To do so, I think you have to build a new class for this admin setting. 

Thanks.